src/staking/manager.ts

```diff
@@ -611,22 +615,42 @@ export class BabylonBtcStakingManager {
    */
   async createProofOfPossession(
     bech32Address: string,
+    stakerBtcAddress: string,
   ): Promise<ProofOfPossessionBTC> {
-    if (!this.btcProvider.signMessage) {
-      throw new Error("Sign message function not found");
+    let sigType: BTCSigType = BTCSigType.ECDSA;
+
+    // For Taproot or Native SegWit addresses, use BIP322 signature scheme for
+    // enhanced security in the proof of possession. For legacy addresses, 
+    // fall back to ECDSA signature scheme.
```

Comment:
```suggestion
    // For Taproot or Native SegWit addresses, use the BIP322 signature scheme
    // in the proof of possession as it uses the same signature type as the regular
    // input UTXO spend. For legacy addresses, 
    // use the ECDSA signature scheme.
```

"enhanced security" might imply that ECDSA is less secure which is not necessarily true.